### ISSUE TITLE
HEAD method not allowed

### ISSUE THREAD

**Issue by livehtml** [sveltejs/kit#3894]
### Describe the bug

I've just run my SvelteKit app in production (adapter node) and periodically gets an error in logs:
405 HEAD Method not allowed

Other methods I can disable with nginx via limit_except, but what about HEAD? Is this method should be supported? How can I enable and correctly handle it?

### Reproduction

Build and run with adapter node

### System Info

```shell
System:
    OS: Linux 5.4 Ubuntu 20.04.3 LTS (Focal Fossa)
    CPU: (8) x64 DO-Premium-Intel
    Memory: 12.50 GB / 15.63 GB
    Container: Yes
    Shell: 5.0.17 - /bin/bash
  Binaries:
    Node: 16.14.0 - /usr/bin/node
    npm: 8.3.1 - /usr/bin/npm
  npmPackages:
    @sveltejs/adapter-node: ^1.0.0-next.68 => 1.0.0-next.68 
    @sveltejs/kit: next => 1.0.0-next.267 
    svelte: ^3.46.4 => 3.46.4
```

### Severity

annoyance

**Comment by antony** [comment]
As far as I'm aware, we don't prevent the `HEAD` method, and this may be a misconfiguration of your server.

Try `npm run preview` locally and see if you can reproduce the issue.

**Comment by livehtml** [comment]
Hmm, may be, but how can fix it?

When I run ```node build``` and then ```curl -i --head http://127.0.0.1:3000```

I've got:
```
HTTP/1.1 405 Method Not Allowed
content-type: text/html
etag: "qkw4jc"
Vary: Accept-Encoding
Date: Mon, 14 Feb 2022 12:53:45 GMT
Connection: keep-alive
Keep-Alive: timeout=5
```

**Comment by livehtml** [comment]
I've just tested on SvelteKit Demo app and found that if we request http://127.0.0.1:3000 — HEAD is allowed (200 OK), but if http://127.0.0.1:3000/todos — got the 405 Method Not Allowed

**Comment by geoffrich** [comment]
Looks like we return a 405 for shadow endpoints if there isn't a handler:

https://github.com/sveltejs/kit/blob/0c6bc25f6f2ef3bad181ac116ed90143e4038ab6/packages/kit/src/runtime/server/page/load_node.js#L386-L394

**Comment by dominikg** [comment]
doesn't look like it https://github.com/sveltejs/kit/blob/master/packages/kit/src/runtime/server/endpoint.js#L46 returns nothing if there is no head handler and https://github.com/sveltejs/kit/blob/master/packages/kit/src/runtime/server/index.js#L167 reads like that returns a 404 then.

**Comment by dominikg** [comment]
working on a fix: 
https://github.com/sveltejs/kit/compare/master...dominikg:fix/default-handlers-for-head

**Comment by bluwy** [comment]
Closing as fixed by #3903
